src/main/kotlin/parts/wisdom/arcraider/visualization/Grid.kt

```diff
@@ -0,0 +1,28 @@
+package parts.wisdom.arcraider.visualization
+
+import parts.wisdom.arcraider.SerializedGrid
+
+data class Coord( val x : Int, val y : Int)
+
+open class Grid(
+    val width: Int,
+    val height: Int,
+    val coords: List<Coord>) {
+
+    constructor(serializedGrid : SerializedGrid) :
+            this(serializedGrid[0].size, serializedGrid[1].size, generateCoords(serializedGrid))
```

Comment:
serializedGrid[1].size is the length of the second row of data, which is not the height. The height is serializedGrid.size.

Reply:
fixed